content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+  
+In the code block shown above, a [partial](#partial) qualifier can be used when declaring SRGs.
+### &lt;Data&gt;
+SRGs can contain fundamental type variables like float, float3, int2, bool, matrices, etc. For each SRG, a Constant Buffer View (CBV) will be automatically created and it will pack all the fundamental type variables declared inside the SRG.
+
+SRGs can contain resource declarations like Texture2D, Texture2DMS, ByteAddressBuffer, etc. In general any HLSL shader resource of the categories:  Shader Resource Views (SRV), Unordered Access Views (UAV), Samplers and Constant Buffer Views (CBV).
+
+SRGs can contain unbounded arrays for resource declarations: SRVs, UAVs, CBVs and Samplers. There are some rules that need to be considered, depending on command line arguments when compiling with **AZSLc**. More on unbounded arrays [here](unbounded-arrays).
+
+SRGs can contain functions.
+
+SRGs can contain structs (declaration & instances). The Shader Build Pipeline and the runtime handle the platform-specific packing rules and alignment.
+
+SRGs can contain instances of classes (but you can not declare classes inside an SRG).  
+  
+**Example**  
+The following example demonstrates declaration of a Shader Resource Group. We have not covered details of what is a `ShaderResourceGroupSemantic`, but suffice to say, it defines the space (aka, descriptor set) index of the constants and resources declared inside the SRG.  
+  
+    ShaderResourceGroupSemantic BindingPerExample
+    {
+        FrequencyId = 0;
+    };
+       
+    ShaderResourceGroup ExampleSRG : BindingPerExample
+    {
+        float4          m_uniformColor;
+        Texture2D       m_texture;
+        Buffer<float>   m_buffer;
+        Sampler m_dynamicSampler;
+        Sampler m_staticSampler
+        {
+            AddressU = Wrap;
+            AddressV = Wrap;
+            AddressW = Wrap;
+            MagFilter = Linear;
+        };
+        
+        // You can define functions inside a ShaderResourceGroup
+        float GetRedColor()
+        {
+            return m_uniformColor.r;
+        }
+    };
+  
+### Scope
+Every SRG defines a scope and the SRG data must be qualified when accessed from out-of-SRG scope code.  
+  
+    float4 color = ExampleSRG::m_uniformColor;
+    color *= ExampleSRG::m_texture.Sample(ExampleSRG::m_staticSampler, float2(0.5, 0.5));
+  
+
+### Sampler variables
+* Dynamic Samplers are declared without a body definition. They have to be set in the runtime. Dynamic Samplers can also be declared as arrays.
+* Static Samplers are declared with a body definition.
+    * Although you can declare Samplers directly as `SamplerState` or `SamplerComparisonState`, AZSL provides a convenient single `Sampler` keyword for both.
+    * Sampler variables will be compiled into regular HLSL `SamplerState` variables, unless a `ComparisonFunc` is defined inside the `Sampler` block, which will be compiled as `SamplerComparisonState`.  
```

Comment:
```suggestion
A dynamic sampler declaration does not contain a body definition because they must be set at runtime. You can also declare a dynamic sampler as an array.

A static sampler declaration contains a body definition. You can declare a sampler and sampler-comparison by using AZSL's `Sampler` keyword, which conviniently encompasses HLSL's `SamplerState` and `SamplerComparisonState` keywords. AZSLc compiles sampler variables into the corresponding HLSL variables: `SamplerState`, or `SamplerComparisonState`, if it contains a `ComparisonFunc` definition.
```

Reply:
Fixed.